src/LHScriptX/FeatureScriptCommands.cpp

```diff
@@ -295,827 +289,626 @@ inline float GetSize(const float size)
-	// " + __FILE__ + ":" + std::to_string(__LINE__) << std::endl;
+	// spdlog::error("LHScriptX: {}:{}: Function {} not implemented.", __FILE__, __LINE__, __func__);
 }
 
-void FeatureScriptCommands::CreateMobileobject(const ScriptCommandContext& ctx)
+void FeatureScriptCommands::CreateMobileObject(glm::vec3 position, int32_t type, int32_t rotation, int32_t scale)
 {
-	Game& game = ctx.GetGame();
-	const auto& params = ctx.GetParameters();
-	auto& island = game.GetLandIsland();
-	auto& registry = game.GetEntityRegistry();
+	auto& registry = Game::instance()->GetEntityRegistry();
 	const auto entity = registry.Create();
-	const auto position = GetHorizontalPosition(params[0].GetString());
-	const auto objectType = params[1].GetNumber();
-	const auto rotation = GetRadians(params[2].GetNumber());
-	float size = GetSize(params[3].GetNumber());
-	auto mobileObjectInfo = static_cast<MobileObjectInfo>(objectType);
 
-	const glm::vec3 pos(position.x, island.GetHeightAt(position), position.y);
-	const glm::mat3 rot = glm::eulerAngleY(rotation);
-	const glm::vec3 scale(size);
-
-	registry.Assign<MobileObject>(entity, mobileObjectInfo);
-	registry.Assign<Transform>(entity, pos, rot, scale);
+	registry.Assign<MobileObject>(entity, static_cast<MobileObjectInfo>(type));
+	registry.Assign<Transform>(entity, position, glm::eulerAngleY(GetRadians(rotation)), glm::vec3(GetSize(scale)));
 }
 
-void FeatureScriptCommands::CreateMobileStatic(const ScriptCommandContext& ctx)
+void FeatureScriptCommands::CreateMobileStatic(glm::vec3 position, int32_t, float, float)
 {
-	std::cout << std::string {} + "Function " + __func__ + " not implemented. " + __FILE__ + ":" + std::to_string(__LINE__)
-	          << std::endl;
+	spdlog::error("LHScriptX: {}:{}: Function {} not implemented.", __FILE__, __LINE__, __func__);
 }
 
-void FeatureScriptCommands::CreateMobileUStatic(const ScriptCommandContext& ctx)
+void FeatureScriptCommands::CreateMobileUStatic(glm::vec3 position, int32_t type, float verticalOffset, float pitch,
+                                                float rotation, float lean, float scale)
 {
-	auto& game = ctx.GetGame();
-	const auto& params = ctx.GetParameters();
-	auto& island = game.GetLandIsland();
-	auto& registry = game.GetEntityRegistry();
+	auto& registry = Game::instance()->GetEntityRegistry();
 	const auto entity = registry.Create();
-	const auto position = GetHorizontalPosition(params[0].GetString());
-	const auto type = params[1].GetNumber();
-	const auto verticalOffset = params[2].GetFloat();
-	float pitch = -params[3].GetFloat();
-	float rotation = -params[4].GetFloat();
-	float lean = -params[5].GetFloat();
-	float size = params[6].GetFloat();
-	auto mobileStaticInfo = MobileStaticInfo(type);
 
-	const glm::vec3 pos(position.x, island.GetHeightAt(position) + verticalOffset, position.y);
-	const glm::mat3 rot = glm::eulerAngleXYZ(pitch, rotation, lean);
-	const glm::vec3 scale(size);
+	registry.Assign<MobileStatic>(entity, MobileStaticInfo(type));
```

Comment:
Forgot to add vertical offset

```suggestion
	position.y += verticalOffset;
	registry.Assign<MobileStatic>(entity, MobileStaticInfo(type));
```

Reply:
:+1: